core/src/test/java/com/linecorp/armeria/client/endpoint/PropertiesEndpointGroupTest.java

```diff
@@ -107,4 +119,119 @@ public void illegalDefaultPort() {
                 .isInstanceOf(IllegalArgumentException.class)
                 .hasMessageContaining("defaultPort");
     }
+
+    @Test
+    public void propertiesFileUpdatesCorrectly() throws Exception {
+
+        final File file = folder.newFile("temp-file.properties");
+        final URL url = file.getParentFile().toURI().toURL();
+        final URLClassLoader classLoader = URLClassLoader.newInstance(new URL[]{url});
+
+        PrintWriter printWriter = new PrintWriter(file);
+        Properties props = new Properties();
+        props.setProperty("serverA.hosts.0", "127.0.0.1:8080");
+        props.store(printWriter, "");
+        printWriter.close();
+
+        final PropertiesEndpointGroup endpointGroupA = PropertiesEndpointGroup.of(
+                classLoader, file.getName(), "serverA.hosts", 80, true);
+
+        await().atMost(20, TimeUnit.SECONDS).until(() -> endpointGroupA.endpoints().size() == 1);
```

Comment:
- If there's no good reason to override the default timeout, please remove `atMost()` calls.
- You could use `untilAsserted()` instead of `until()`:

  ```java
  await().untilAsserted(() -> assertThat(endpointGroupA.endpoints()).hasSize(1));
  ```